fix(template/vue): remove unused import Vue

theruggerdev opened the pull request:
Resolves the playground compile error (no-unused-var) in vue example

Fixes teambit/bit #2355

Additionally, for general and test readability:
- Converted joined arrays to template literals
- Tabs/spaces in templates replaced with empty strings in test

Commit:
fix(template/vue): removed unused var Vue

Resolves the playground compile error for the vue example template

Fixes teambit/bit #2355

Additionally, for easier interpretation:

- Refactored the test and replaced joined arrays with template literals
- Replaced tabs and double-spaces in test due to inconsistent parsing

KutnerUri commented:
hey!
Finally finished our new homepage and got around to this code review.
See notes inline

KutnerUri commented on components/template/vue/vue-template.spec.ts:
```diff
@@ -1,41 +1,22 @@
-		'</style>',
-	].join('\n');
+	result.files['Index.vue'] = result.files['Index.vue']
+		.replace(/\t/g, '').replace(/  /g, ''); // tabs and double-spaces removed for comparison
```
you are hiding bugs!
I know it is difficult to compare with whitespaces, but this is what actually goes on the screen. 

Here is what I got when I ran it locally:
### expected:
```vue
<template>
  <CodeGenerator/>
</template>

<script>
  import CodeGenerator from '@bit/bit.javascript.raw.code-generator';

  export default (
    {
      data () {
        return {
          var1: 'World'
        }
      },
      components: {
        CodeGenerator
      }
    }
  )
</script>

<style scoped>

</style>
```

### actual
```vue
<template>
                <CodeGenerator/>
        </template>

        <script>
                import CodeGenerator from '@bit/bit.javascript.raw.code-generator';

        export default (
                {
                                data () {
                                        return {
                                                var1: 'World'
                                        }
                                },
                                components: {
                                        CodeGenerator
                                }
                        }
        )
        </script>

        <style scoped>

        </style>
```

KutnerUri replied:
you can see it more easily by adding ```console.log()```

paulrenshaw replied:
It is difficult to compare with whitespaces, since different environments appear to interpret spaces/tabs differently in different files. My tests worked and console logging output (before replacing anything) showed \t in place of tabs from the generated code, but spaces in the snapshot code.

In the new commit, I've only replaced spaces in the snapshot with tabs (expecting tabs in the generated code to be correct without modification.

I suggest that what should actually happen is that the code is generated without tabs and compared to a snapshot without tabs, then prettified by the editor at runtime (e.g. the example editor on bit.dev should put the tabs in, as would any personal editor based on the editor's settings for the number of spaces per tab). Since tabs/spaces are not required for javascript to be interpreted, they should not be included in code used for testing.

KutnerUri replied:
looks good. I cloned it locally and the test pass. It also worked when I inlined the tabs.

As for prettifier, yes, it looks like the correct way to go, but good luck getting a light vue prettifier that works in a browser! haha.

KutnerUri commented on components/template/vue/vue-template.ts:
```diff
@@ -4,7 +4,7 @@ import {
 import { CodeGenerator } from '@bit/bit.javascript.raw.code-generator';
 
-export const mainFile = 'index.vue';
+export const mainFile = 'Index.vue';
```
Please leave lowercase, as the Vue Capsule imports `./src/index.vue` and may break.

Plus, mixed casing may cause troubles in Windows, so we always prefer lower case `kebab-case.tsx` for files

paulrenshaw replied:
done

KutnerUri replied:
nice

KutnerUri commented on components/template/vue/vueIndex.ts:
```diff
@@ -0,0 +1,25 @@
+export default `<template>
```
I like this a lot!

Please move to a ```/snapshots/``` folder.
For a minute there I thought the component had a second index file!

KutnerUri replied:
also I'd mark this as a test file, but I tried it locally and the test gets stuck on 'running specs' 😆
@GiladShoham

paulrenshaw replied:
moved

KutnerUri replied:
nice!

KutnerUri reviewed (changes requested):
thanks for the PR!

paulrenshaw commented:
Hi @KutnerUri, how do I go about making the changes? Do I make another pull request or is there some other method to update this one?

Commit:
fix(template/vue): replace literals with joined arrays

Implements requested changes to teambit/example-templates Pull Request #1

Test also cleaned up and designed to test default export

Potential issues:
- double-spaces in snapshot replaced with tabs, not sure how to fix

Commit:
fix(template/vue): use kebab-case for filenames

KutnerUri commented:
ok looks good, tests pass, I will merge soon and deploy.

I hope you are safe wherever you are! 😃
